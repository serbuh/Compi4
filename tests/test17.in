// Complex boolean expressions
bool a = (3<7) and (true or true or true);
bool b = (300 K == 27 C);
bool c = true or false or false or false;
bool d = (22*55 + 8)!=12 and true;

if(a) {
  print "Pass 1";
} else {
  print "This shouldn't be printed #1";
}

if(b) {
  print "Pass 2";
}

if(not c) {
  print "This shouldn't be printed #2";
} else {
  print "Pass 3";
}

if(d) {
  print "Pass 4";
} else {
  print "This shouldn't be printed #3";
}

if(false or false or true) {
  print "Pass 5";
} else {
  print "This shouldn't be printed #4";
}

if(16>15 or (200 K != 150 K and 16 C >= 16 C)) {
  print "You're awesome!";
} else {
  print "This shouldn't be printed #5";
}

if (true or (true and (false or true or false and (2>1)))) {
  print "You're awesome 2!";
} else {
  print "This shouldn't be printed #6";
}

if((5*6*7 + 2) > 1000) {
  print "This shouldn't be printed #7";
} else {
  print "Hooray!";
}
