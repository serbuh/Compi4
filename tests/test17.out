Pass 1
Pass 2
Pass 3
Pass 4
Pass 5
You're awesome!
You're awesome 2!
Hooray!
